webapp/src/types/validators.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {Question, SurveyQuestionsConfig, SurveyResponse} from 'types/plugin';
+
+export function validateSurveyQuestionsConfig(obj: unknown): asserts obj is SurveyQuestionsConfig {
```

Comment:
Interesting. I've never seen the `asserts` keyword before. This seems like an interesting alternative to regular type predicates.

I wonder if these are something we should introduce to the web app in testing environments to ensure that TS objects are actually the type that they say they are. I occasionally find values where the TS types completely don't match what we say they are, so it would be nice if we could have more certainty around them by using assertions sometimes

Reply:
Yeah, it might be nice, but we need a library/tool to auto-generate all this code. Writing all this manually is PITA